packages/api-contract/src/base/util.ts

```diff
@@ -44,3 +44,23 @@ export function encodeSalt (salt: Uint8Array | string | null = randomAsU8a()): U
       ? compactAddLength(u8aToU8a(salt))
       : EMPTY_SALT;
 }
+
+export function expandNs <T> (ns: Namespaced<T>, { path }: AbiMessage, call: T): T {
+  if (path.length > 1) {
+    for (let i = 0; i < path.length - 2; i++) {
+      ns = ns[path[i]] = {} as Namespaced<T>;
+    }
+  }
+
+  ns[path[path.length - 1]] = call;
+
+  return call;
+}
+
+export function expandConstructors <ApiType extends ApiTypes, R extends ISubmittableResult> (constructors: AbiMessage[], ns: Namespaced<BlueprintDeploy<ApiType>>, tx: MapConstructorExec<ApiType>, creator: (constructorOrId: AbiConstructor | string | number, options: BlueprintOptions, params: unknown[]) => SubmittableExtrinsic<ApiType, R>): void {
```

Comment:
At the moment ink! doesn't plan to support namespaced constructors, so maybe we can remove that